### PR TITLE
[Metrics] added wiki_split metrics

### PR DESCRIPTION
**Pull request by bhadreshpsavani**
Fixes: #2606

This pull request adds combine metrics for the wikisplit or English sentence split task

Reviewer: @patrickvonplaten

### CODE REVIEW

**Comment by hustlen0mics** [comment]
Too

**Comment by hustlen0mics** [comment]
> Too

Flo

File: metrics/wiki_split/wiki_split.py [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+sentences, and a list of lists of reference sentences.
+Args:
+    sources: list of source sentences where each sentence should be a string.
+    predictions: list of predicted sentences where each sentence should be a string.
+    references: list of lists of reference sentences where each sentence should be a string.
+Returns:
+    sari: sari score
+    sacrebleu: sacrebleu score
+    exact: exact score
+Examples:
```
Comment (patrickvonplaten) [inline metrics/wiki_split/wiki_split.py:67]:
```suggestion

Examples:
```

File: metrics/wiki_split/wiki_split.py [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+
+    return white_space_fix(remove_articles(remove_punc(lower(s))))
+
+
+def compute_exact(a_gold, a_pred):
+    return int(normalize_answer(a_gold) == normalize_answer(a_pred))
+
+
+def compute_em(predictions, references):
+    scores = [any([compute_exact(ref, pred) for ref in refs]) * 1 for pred, refs in zip(predictions, references)]
```
Comment (patrickvonplaten) [inline metrics/wiki_split/wiki_split.py:103]:
```suggestion
    scores = [any([compute_exact(ref, pred) for ref in refs]) for pred, refs in zip(predictions, references)]
```

Nit don't think you need to convert boolean to int here - it's done automatically when using `sum(...)`

**Review by patrickvonplaten — APPROVED** [review]
Looks very clean to me! Just left some nits. 

@lhoestq - for reference, the task of sentence splitting is usually evaluated by three metrics:

- bleu
- exact match
- sari 

as is done in this paper: https://arxiv.org/abs/1907.12461 for example.

File: metrics/wiki_split/wiki_split.py [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+
+_DESCRIPTION = """\
+WIKI_SPLIT is the combination of three metrics SARI, EXACT and SACREBLEU
+It can be used for evaluate performance performance of generated sentences
+"""
+
+
+_KWARGS_DESCRIPTION = """
+Calculates sari score (between 0 and 100) given a list of source and predicted
+sentences, and a list of lists of reference sentences.
```
Comment (lhoestq) [inline metrics/wiki_split/wiki_split.py:58]:
```suggestion
sentences, and a list of lists of reference sentences. It also computes the BLEU score as well as the exact match score.
```

File: metrics/wiki_split/wiki_split.py [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+    address = "Belgium, Brussels",
+    publisher = "Association for Computational Linguistics",
+    url = "https://www.aclweb.org/anthology/W18-6319",
+    pages = "186--191",
+}
+"""
+
+_DESCRIPTION = """\
+WIKI_SPLIT is the combination of three metrics SARI, EXACT and SACREBLEU
+It can be used for evaluate performance performance of generated sentences
```
Comment (lhoestq) [inline metrics/wiki_split/wiki_split.py:52]:
```suggestion
It can be used to evaluate the quality of machine-generated texts.
```

**Review by lhoestq — APPROVED** [review]

**Comment by lhoestq** [comment]
Looks all good to me thanks :)

**Pull request merged**